### PR TITLE
Makes sure we return the dependent bundles in the right order.

### PR DESCRIPTION
**Pull request by MiguelMadero**

### CODE REVIEW

File: tests/acceptance/services/lazy-loader-test.js [hunk 1]
```diff
@@ -151,3 +151,39 @@ test('dependsOn doesnt break for circular dependencies', function(assert) {
...
+    name: '2',
+    dependsOn: ['4', '3'],
+  }, {
+    name: '3',
+    dependsOn: ['4'],
+  }, {
+    name: '4'
+  }]);
+
+  var assets = namesForBundles(service.getDependentBundlesForBundle(service.getBundleByName('1')));
```
Comment (jcollins1991-zenefits) [inline tests/acceptance/services/lazy-loader-test.js]: seems like this should be two separate tests. also "dependsOn respects the order" as the name is confusing when you swap the order of the dependsOn but the results don't change

**Comment by jcollins1991-zenefits** [comment]
this primarily matters in instances where you're working w/ globals instead of modules, right ..?

**Comment by jcollins1991-zenefits** [comment]
LGTM

**Comment by MiguelMadero** [comment]
Thanks for the feedback I updated the test name. 

> this primarily matters in instances where you're working w/ globals instead of modules, right ..?

Mostly globals, but more generally, anything outside of modules. For example, if we had inline code at the bottom of the file that requires a module that depends on a module of another package.

**Comment by MiguelMadero** [comment]
released as 0.3.6

**Pull request merged**